File: src/main/scala/com/github/mauricio/async/db/postgresql/parsers/ReturningMessageParser.scala
package com.github.mauricio.postgresql.parsers

import com.github.mauricio.postgresql.messages.backend.Message
import org.jboss.netty.buffer.ChannelBuffer

/**
 * User: Maurício Linhares
 * Date: 3/12/12
 * Time: 11:36 PM
 */

class ReturningMessageParser( val message : Message ) extends Decoder {
  def parseMessage(buffer: ChannelBuffer): Message = {
    this.message
  }
}

File: postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/parsers/ReturningMessageParser.kt
package com.github.jasync.sql.db.postgresql.parsers

import com.github.jasync.sql.db.postgresql.messages.backend.BindComplete
import com.github.jasync.sql.db.postgresql.messages.backend.CloseComplete
import com.github.jasync.sql.db.postgresql.messages.backend.EmptyQueryString
import com.github.jasync.sql.db.postgresql.messages.backend.NoData
import com.github.jasync.sql.db.postgresql.messages.backend.ParseComplete
import com.github.jasync.sql.db.postgresql.messages.backend.ServerMessage
import io.netty.buffer.ByteBuf

enum class ReturningMessageParser(val message: ServerMessage) : MessageParser {

    BindCompleteMessageParser(BindComplete),
    CloseCompleteMessageParser(CloseComplete),
    EmptyQueryStringMessageParser(EmptyQueryString),
    NoDataMessageParser(NoData),
    ParseCompleteMessageParser(ParseComplete);

    override fun parseMessage(buffer: ByteBuf): ServerMessage = this.message
}
